modules/contracts/adapters/dex/src/dex_tester.rs

```diff
@@ -0,0 +1,474 @@
+use crate::{interface::DexAdapter, msg::DexInstantiateMsg, DEX_ADAPTER_ID};
+use abstract_client::{AbstractClient, ClientResolve, Environment};
+use abstract_core::{
+    adapter,
+    objects::{
+        module::{ModuleInfo, ModuleVersion},
+        pool_id::PoolAddressBase,
+        AnsAsset, AssetEntry, LpToken, PoolMetadata,
+    },
+};
+use abstract_dex_standard::{ans_action::DexAnsAction, msg::DexExecuteMsg};
+use abstract_interface::{Abstract, AdapterDeployer, DeployStrategy, ExecuteMsgFns, VCExecFns};
+use cosmwasm_std::{coins, Decimal, Uint128};
+use cw_asset::AssetInfoUnchecked;
+use cw_orch::{anyhow, environment::MutCwEnv, prelude::*};
+
+pub struct DexTester<Chain: MutCwEnv, Dex: MockDex> {
+    pub abstr_deployment: AbstractClient<Chain>,
+    pub dex_adapter: DexAdapter<Chain>,
+    pub dex: Dex,
+    pub lp_asset: AssetInfoUnchecked,
+}
+
+impl<Chain: MutCwEnv, Dex: MockDex> DexTester<Chain, Dex> {
+    pub fn new(abstr_deployment: AbstractClient<Chain>, dex: Dex) -> anyhow::Result<Self> {
+        // Re-register dex, to make sure it's latest
+        let _ = abstr_deployment
+            .version_control()
+            .remove_module(ModuleInfo::from_id(
+                DEX_ADAPTER_ID,
+                ModuleVersion::Version(crate::contract::CONTRACT_VERSION.to_owned()),
+            )?);
+        let dex_adapter = DexAdapter::new(DEX_ADAPTER_ID, abstr_deployment.environment());
+        dex_adapter.deploy(
+            crate::contract::CONTRACT_VERSION.parse()?,
+            DexInstantiateMsg {
+                recipient_account: 0,
+                swap_fee: Decimal::permille(3),
+            },
+            DeployStrategy::Force,
+        )?;
+
+        let lp_asset = {
+            let (pool, pool_metadata, lp_asset) = dex.create_pool()?;
+            let abstr = Abstract::load_from(abstr_deployment.environment())?;
```

Comment:
Are we sure we don't want to expose abstract in the client? xd

Reply:
ah, I didn't need to use it though. forgot we have ans exposed, haha https://docs.rs/abstract-client/latest/abstract_client/struct.AbstractClient.html#method.name_service